### PR TITLE
Replace bsd x86 exec payload and DRY up osx/bsd code

### PR DESCRIPTION
**Pull request by joevennix**
This should close out #5124. This replaces the x86 bsd exec payload with osx's, which is a bit shorter, and allows any binary + set of args to be passed straight to execve. I've also merged all duplicate shellcode between BSD and OSX on x86 into a BSD mixin, `Msf::Payload::Bsd::X86`.

Additionally, this removes two prepends stubs, `setresgid` and `setresuid` from the OSX payload mixin's datastore options. These syscalls have never been present on OSX.
#### Verification
- [x] Generate a BSD x86 exec payload with prepend flags set, ensure it works as expected
  
  ```
  $ ./msfpayload bsd/x86/exec CMD='/bin/echo 1 2 3' PrependSetreuid=1 PrependSetuid=1 PrependSetregid=1 PrependSetgid=1 PrependSetresuid=1 Prependsetresgid=1 AppendExit=1 LHOST=192.168.0.4 LPORT=5555 X > /tmp/bsd.bin
  $ chmod +x /tmp/bsd.bin
  $ /tmp/bsd.bin
  1 2 3
  $ sudo su
  # kldload dtrace 
  # kldload dtraceall
  # dtruss /tmp/bsd.bin
  ..
  setresuid(0x0, 0x0, 0x0)         = 0 0
  setreuid(0x0, 0x0, 0x0)      = 0 0
  setuid(0x0, 0x0, 0x0)        = 0 0
  setresgid(0x0, 0x0, 0x0)         = 0 0
  setregid(0x0, 0x0, 0x0)      = 0 0
  setgid(0x0, 0x0, 0x0)        = 0 0
  execve(0x4000ED, 0x7FFFFFFFE990, 0x0)        = 0 0
  ```
- [x] Now do the same for the osx x86 payload:
  
  ```
  ./msfpayload osx/x86/exec CMD='/bin/echo 1 2 3' PrependSetreuid=1 PrependSetuid=1 PrependSetregid=1 PrependSetgid=1 AppendExit=1 LHOST=192.168.0.4 LPORT=5555 X > /tmp/osx.bin
  $ chmod +x /tmp/osx.bin
  $ /tmp/osx.bin
  1 2 3
  $ sudo dtruss /tmp/osx.bin
  ..
  setreuid(0x0, 0x0, 0xBFFFDE94)           = 0 0
  setuid(0x0, 0x0, 0xBFFFDE94)             = 0 0
  psynch_cvclrprepost(0x0, 0x0, 0x0)               = 0 0
  setregid(0x0, 0x0, 0x0)          = 0 0
  setgid(0x0, 0x0, 0x0)            = 0 0
  ..
  ```

### CODE REVIEW

File: modules/auxiliary/dos/osx/darwin_nuke_icmp.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+##
```
Comment (wvu-r7) [inline modules/auxiliary/dos/osx/darwin_nuke_icmp.rb]: *One of these things is not like the others
One of these things just doesn't belong
Can you tell which thing is not like the others
By the time I finish my song?*

Comment (joevennix) [inline modules/auxiliary/dos/osx/darwin_nuke_icmp.rb]: hahaha thank you. `git commit .` beats me again.

**Comment by joevennix** [comment]
Force pushed to fixed accidental inclusion of defunct module in my last commit.

**Comment by wvu-r7** [comment]
https://github.com/rapid7/metasploit-framework/commit/13da15e434c167709dfa43214eba7a94daadda5d

**Pull request merged**